services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/client_api.py

```diff
@@ -269,6 +273,30 @@ async def get_entrypoint_container_name(
             log_httpx_http_error(url, "GET", traceback.format_exc())
             raise
 
+    async def restart_containers(self, dynamic_sidecar_endpoint: str) -> None:
+        """
+        runs docker-compose stop and docker-compose start in succession
+        resulting in a container restart without loosing state
+        """
+        url = get_url(
+            dynamic_sidecar_endpoint, f"/{self.API_VERSION}/containers:restart"
+        )
+        try:
+            async with httpx.AsyncClient(
```

Comment:
I am not convinced that creating a client instance for every request is the way to go (everybody recommends otherwise. check e.g. [httpx](https://www.python-httpx.org/advanced/#client-instances) and [aiohttp-client](https://docs.aiohttp.org/en/stable/http_request_lifecycle.html) ). We have discussed it multiple times ... i guess we need to reach a conclusion and use the best approach in all services

Reply:
Previously for some reason this made sense to me. Just refactored it to use a shared client.